frame/composable-support/src/collections/vec/bounded/bi_bounded_vec.rs

```diff
@@ -0,0 +1,526 @@
+//! https://github.com/ergoplatform/bounded-vec/issues/13
+use codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+#[cfg(feature = "std")]
+use serde::{Deserialize, Serialize};
+use sp_std::{
+	convert::{TryFrom, TryInto},
+	slice::{Iter, IterMut},
+	vec::{self, Vec},
+};
+
+/// Non-empty Vec bounded with minimal (L - lower bound) and maximal (U - upper bound) items
+/// quantity
+#[derive(PartialEq, Eq, Debug, Clone, Hash, PartialOrd, Ord, Encode, TypeInfo)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize), serde(transparent))]
+pub struct BiBoundedVec<T, const L: usize, const U: usize>
+// enable when feature(const_evaluatable_checked) is stable
+// where
+//     Assert<{ L > 0 }>: IsTrue,
+{
+	inner: Vec<T>,
+}
+
+impl<T: Decode, const L: usize, const U: usize> Decode for BiBoundedVec<T, L, U> {
+	fn decode<I: codec::Input>(input: &mut I) -> Result<Self, codec::Error> {
+		let raw: Vec<_> = Vec::decode(input)?;
+		BiBoundedVec::try_from(raw).map_err(|_err| codec::Error::from("Input out of bounds"))
+	}
+}
+
+impl<T: Decode + MaxEncodedLen, const L: usize, const U: usize> MaxEncodedLen
+	for BiBoundedVec<T, L, U>
+{
+	fn max_encoded_len() -> usize {
+		U * T::max_encoded_len()
+	}
+}
+
+/// BiBoundedVec errors
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub enum BiBoundedVecOutOfBounds {
+	/// Items quantity is less than L (lower bound)
+	LowerBoundError {
+		/// L (lower bound)
+		lower_bound: usize,
```

Comment:
Maybe rename to `expected` and `received`?

Reply:
i do not want to go far from original crate. hope they will so substrate at some point. that crate in repo of some blockchain.